src/pages/GanacheDetails.ts

```diff
@@ -0,0 +1,370 @@
+  }
+
+  /**
+   * This action is responsible for getting all project information from the Network Treeview
+   *
+   * @param projectView - represents the project instance coming from Network Treeview
+   */
+  private async getProjectInformation(projectView: ProjectView): Promise<TProjectInformation> {
+    // Gets the local project and network node concerning to the Treeview Item
+    const localProject = projectView.extensionItem as LocalProject;
+    const networkNode = projectView.extensionItem.getChildren()[0] as LocalNetworkNode;
+
+    // Returns the object with all project information formatted
+    return {
+      projectView: projectView,
+      label: localProject.label,
+      options: localProject.options,
+      port: networkNode.port,
+      networkId: networkNode.networkId as string,
+    };
+  }
+
+  /**
+   * This action is responsible for checking if the Ganache instance is running.
+   * If the instance is not running, it starts.
+   *
+   * @param projectInformation - represents all formatted project information coming from a ProjectView
+   */
+  private async isGanacheRunning(projectInformation: TProjectInformation): Promise<boolean> {
+    // Checks the port status
+    const portStatus = await GanacheService.getPortStatus(projectInformation.port);
+    let isRunning = false;
+
+    // Checks the Ganache status instance and take the proper action
+    switch (portStatus) {
+      case GanacheService.PortStatus.FREE:
+        await GanacheCommands.startGanacheCmd(
+          async () => await GanacheCommands.selectGanachePortAndOptions(projectInformation.projectView)
+        );
+        isRunning = true;
+        break;
+      case GanacheService.PortStatus.GANACHE:
+        isRunning = true;
+        break;
+      case GanacheService.PortStatus.NOT_GANACHE:
+        isRunning = false;
+        break;
+    }
+
+    // Returns whether the instance is running or not
+    return isRunning;
+  }
+
+  /**
+   * This action is responsible for getting all provider information,
+   * including block, gas, transactions and others.
+   *
+   * @param projectInformation - represents all formatted project information coming from a ProjectView
+   */
+  private async getProviderInformation(projectInformation: TProjectInformation): Promise<TProviderInformation> {
+    // Set the web3 provider
+    const web3 = await this.setWeb3(projectInformation);
+
+    // Gets the last block number
+    const blockNumber = await web3.eth.getBlockNumber();
+
+    // Gets the information concerning the block
+    const {gasLimit} = await web3.eth.getBlock(blockNumber);
+
+    // Returns the provider information
+    return {
+      currentBlock: blockNumber,
+      gasLimit: gasLimit,
+      networkId: projectInformation.networkId,
+      rpcServer: `${Constants.networkProtocols.http}${Constants.localhost}:${projectInformation.port}`,
+      gasPrice: await web3.eth.getGasPrice(),
```

Comment:
Related to typings, `await web3.eth.getGasPrice()` returns a `string`, but `gasPrice` is defined as `number`. Is this intentional?